src/main/java/com/example/hop_oasis/controller/RecommendationsController.java

```diff
@@ -0,0 +1,53 @@
+package com.example.hop_oasis.controller;
+
+import com.example.hop_oasis.convertor.RecommendationsMapper;
+import com.example.hop_oasis.dto.RecommendationsDto;
+import com.example.hop_oasis.handler.ValidItemType;
+import com.example.hop_oasis.service.RecommendationsService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/recommendations")
+@Tag(name = "Recommendations", description = "APIs for getting product's recommendation")
+public class RecommendationsController {
+
+    private final RecommendationsService recommendationsService;
+    private final RecommendationsMapper mapper;
+
+    public RecommendationsController(RecommendationsService recommendationsService,
+                                     RecommendationsMapper mapper) {
+        this.recommendationsService = recommendationsService;
+        this.mapper = mapper;
+    }
+
+    @GetMapping("/for-cart")
+    @Operation(summary = "Get recommendations based on what product items are already in the cart")
+    public ResponseEntity<RecommendationsDto> forCart(
+            @Parameter(description = "Cart id", required = true)
+            @RequestParam("cartId") Long cartId) {
+
+        return ResponseEntity.ok(
+                mapper.toDto(recommendationsService.getForCart(cartId)));
+    }
+
+    @GetMapping("/for-product")
+    @Operation(summary = "Get recommendations based on product")
+    public ResponseEntity<RecommendationsDto> forCart(
+            @Parameter(description = "product id", required = true)
+            @RequestParam("productId") Long productId,
+
+            @Parameter(description = "Product type, for now it's one of BEER, CIDER, SNACK, PRODUCT_BUNDLE",
+                    required = true)
+            @RequestParam("itemType") @ValidItemType String itemType) {
```

Comment:
why don't just use ItemType enum here?

Reply:
I think I did it so that validation exception will be thrown if itemType is incorrect also I wanted to practice in creation of custom validators